### PR TITLE
Setup benchmarks stats and graphs

### PR DESCRIPTION
**Pull request by storozhukBM**
For more details look at: https://github.com/uber-go/ratelimit/pull/85#discussion_r889650431

### CODE REVIEW

**Comment by storozhukBM** [comment]
@rabbbit 
Hmmm, the atomic limiter test has failed even though I didn't make any changes

File: tools/go.mod [hunk 1]
```diff
@@ -4,8 +4,10 @@ go 1.14
```
Comment (rabbbit) [inline tools/go.mod:4]: I'll bump the go version here to get rid of the transient dependencies.

**Comment by rabbbit** [comment]
> @abhinav @rabbbit Hmmm, the atomic limiter test has failed even though I didn't make any changes

Yeah, we've seen this happen (year+ ago, especially during codegen generation), but didn't have time to debug. There's some unknown flakiness, maybe this is a good push for me to actually take a look :)

**Comment by rabbbit** [comment]
Thanks!

**Comment by rabbbit** [comment]
Hey @storozhukBM I'm getting 

image

when running benchmarks on main. I didn't look into this, I assumed you'd know what's going on :)

**Comment by storozhukBM** [comment]
@rabbbit yes, `benchart` requires go 1.18 in its go mod, if you want I can lower to 1.17, but it is better to run benchmarks on latest Go version

**Comment by rabbbit** [comment]
> @rabbbit yes, `benchart` requires go 1.18 in its go mod, if you want I can lower to 1.17, but it is better to run benchmarks on latest Go version

Hm. Technically we want to support two latest versions .. but I guess for benchmarks it's okay. Trying to fix with https://github.com/uber-go/ratelimit/pull/88 - not sure how to test actions without landing them first 🤷

**Comment by storozhukBM** [comment]
I'm not sure what the point of running benchmarks on Github actions is since they run on some virtual machines with unpredictable performance that can vary from run to run.

**Comment by rabbbit** [comment]
It's executed only when new commit is pushed on main, which is probably
around once a year on average :)

I won't pay too much attention to it, but I thought it's something "nice to
have" to perhaps take a look and see in the UI. Also, it tests whether the
benchmarks are runnable.

**Pull request merged**